ClusteringAlgs/HierarchyCluster/BlossomCluster.py

```diff
@@ -33,19 +69,23 @@ def cluster_once(clusters: List[Set[int]], matrix: np.ndarray) -> Tuple[List[Set
     @staticmethod
     def cluster(matrix: np.ndarray, cluster_size: int) -> List[Set[int]]:
         n = matrix.shape[0]
-        k = cluster_size
         clusters = [{i} for i in range(n)]
         while True:
+            matches_init = [(i, i) for i in range(matrix.shape[0])]
             clusters, matches = BlossomCluster.cluster_once(clusters, matrix)
-            k /= 2
-            if round(k) <= 1:
-                break;
+            if is_equal(matches, matches_init):
```

Comment:
is_equal is redundant. matches length in less than matches_init or it's equal if there are no matches.